### PR TITLE
Separate load breaches script

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: scripts/load-breaches.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  try {
+    const breachesJSON = JSON.parse(response.body);
+
+    for (const breach of breachesJSON) {
+      await DBUtils.createBreach(breach.Name, breach);
+    }
+  } catch (error) {
+    console.error(error);
+    // We can `process.exit()` here since it's a CLI script.
+    // eslint-disable-next-line no-process-exit
```
[the remark this answers is not in the document]
Comment (pdehaan) [reply scripts/load-breaches.js]: Another potential tweak is to allow `console.warn()` and `console.error()` by tweaking the following rule:

```js
"no-console": ["warn", {allow: ["warn", "error"]}],
```

This reduces it to 9 **warnings**.

But I can tweak all these configs and rules in a separate PR since it's unrelated to this specific issue.

[the remark this answers is not in the document]
Comment (pdehaan) [reply scripts/load-breaches.js]: You can ignore most of this noise. Submitted PR #109; _"Resolve some ESLint console warnings"_, which reduces everything to a much more reasonable 3 warnings:

```sh
$ npm run lint:js

> blurts-server@0.0.1 lint:js /Users/pdehaan/dev/github/mozilla/blurts-server
> eslint .


/Users/pdehaan/dev/github/mozilla/blurts-server/routes/oauth.js
  45:5  warning  Unexpected 'todo' comment  no-warning-comments

/Users/pdehaan/dev/github/mozilla/blurts-server/routes/user.js
  28:3  warning  Unexpected 'todo' comment     no-warning-comments
  29:3  warning  Unexpected console statement  no-console

✖ 3 problems (0 errors, 3 warnings)
```

File: db/utils.js [hunk 2]
```diff
@@ -18,17 +18,15 @@ Model.knex(knex);
 const DBUtils = {
   async createBreach(name, meta) {
     try {
-      return await Breach
-        .query()
-        .insert({ name, meta });
+      const insertedBreach = await Breach.query().insert({ name, meta });
```
Comment (nhnt11) [inline db/utils.js]: What's the reason for this change?

File: db/utils.js [hunk 3]
```diff
@@ -18,17 +18,15 @@ Model.knex(knex);
...
     } catch(e) {
+      console.error(e);
       if (e.code && e.code === "23505") {
         // Duplicate error, silently log.
-        console.log(`Duplicate breach: ${name}`);
+        console.error(`Duplicate breach: ${name}`);
         return;
       }
-
-      throw e;
```
Comment (nhnt11) [inline db/utils.js]: What's the reason for removing this here? IMO the database error should be propagated and handled by consumers of this function.

File: scripts/load-breaches.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+async function handleBreachesResponse(response) {
+  try {
+    const breachesJSON = JSON.parse(response.body);
+
+    for (const breach of breachesJSON) {
+      await DBUtils.createBreach(breach.Name, breach);
+    }
+  } catch (error) {
+    console.error(error);
```
Comment (nhnt11) [inline scripts/load-breaches.js]: I don't think this will ever be reached, unless there is a JSON parsing error, since the `throw` was removed from `DBUtils.createBreach`.

File: tests/fixtures/make-breach-with-emails.js [hunk 5]
```diff
@@ -1,60 +1,64 @@
...
+    await DBUtils.createBreach("AllMusic", {
+      Name: "AllMusic",
+      BreachDate: "2015-012-06",
+      DataClasses: ["Email addresses", "IP addresses", "Passwords", "Usernames", "Website activity"],
+      PwnCount: 1436486,
+    });
+  }
   for (const sB of sampleBreaches) {
-    await DBUtils.createBreach(sB.name, sB.meta);
+    await DBUtils.getBreachByName(sB.name);
```
Comment (nhnt11) [inline tests/fixtures/make-breach-with-emails.js]: Why are we getting a breach and not doing anything with it?

Comment (nhnt11) [reply tests/fixtures/make-breach-with-emails.js]: I'm not sure I understand what the purpose of this code is anymore. Why remove the meta data from the other sample breaches? Why try to add the extra emails for LinkedIn/Adobe when they may not exist? (We only have an option to add the all music breach?)

File: db/utils.js (hunk 2 above)
Comment (groovecoder) [reply db/utils.js]: Oh, I think that's left-over when I was debugging by `console.log`ging the `insertedBreach` value. I can change it back.

File: tests/fixtures/make-breach-with-emails.js (hunk 5 above)
Comment (groovecoder) [reply tests/fixtures/make-breach-with-emails.js]: Hmm ... yeah I'm sorta mangling these scripts together. 😢 I should add a call to `load-breaches` from here, or maybe rename this to `add-breached-emails.js` ...

Basically, I'm changing this `make-breach-with-emails.js` to only add the breached emails. It adds the `test[1-3]@test.com` emails to the `LinkedIn`, `Adobe`, and `AllMusic` breaches as before, but it now relies on the new `load-breaches` script to create those breaches with the *real* data from HIBP.

I should also probably move the `--createAMBreach` code into the `load-breaches` script.

Let me try to clean this up a bit ...

Comment (groovecoder) [reply tests/fixtures/make-breach-with-emails.js]: Okay, I un-mangled this logic and moved `load-breaches` and `add-breached-emails` into `scripts/` (per https://github.com/mozilla/blurts-server/pull/109#pullrequestreview-120666384) and updated the `README` with instructions on how to use them.

Needs a new r?

**Comment by groovecoder** [comment]
@nhnt11 - is this good to go now?
